master/pyext/src/validation/sas.py

```diff
@@ -1,692 +1,756 @@
+                if response.status_code == 200:
+                    output.append('True')
+
+        return output
+
     def get_sascif_file(self):
         '''
         get data from SASCIF files
         '''
         for code in self.get_SASBDB_code():
             if 'None' not in str(code):
-                url_f=self.saslink+code+'.sascif'
-                response=requests.get(url_f);
-                if response.status_code!=200:
-                    print ("Error....unable to fetch data from SASBDB, please check the entry ID")
-                with open (code+'.sascif', 'w') as f:
-                    f.write(response.text);
+                url_f = self.saslink+code+'.sascif'
+                response = requests.get(url_f)
+                if response.status_code != 200:
+                    print(
+                        "Error....unable to fetch data from SASBDB, please check the entry ID")
+                with open(code+'.sascif', 'w') as f:
+                    f.write(response.text)
 
-    def get_all_sascif(self,sasbdb)->list:
+    def get_all_sascif(self, sasbdb) -> list:
         '''
         get a list of all lines in a SASCIF file
         '''
         if 'None' not in str(sasbdb):
-            file=open(sasbdb+'.sascif','r')
-            all_lines=[data.strip().split() for indx,data in enumerate(file.readlines())]
+            file = open(sasbdb+'.sascif', 'r')
+            all_lines = [data.strip().split()
+                         for indx, data in enumerate(file.readlines())]
         return all_lines
 
-    def get_intensities(self)->dict:
+    def get_intensities(self) -> dict:
         '''
         get intensity data from SASCIF file
         if SASCIF file is not present, this information will not be present/used in the report
-        JSON file typically has raw data 
+        JSON file typically has raw data
         '''
         self.get_sascif_file()
-        Int_dict={}
+        Int_dict = {}
         for code in self.clean_SASBDB_code():
-            #Int_dict={}
-            all_lines=self.get_all_sascif(code)
-            data={}
-            for indx,sascifline in enumerate(all_lines):
-                if len(sascifline)<2 and len(sascifline)>0 and 'scan_intensity' in sascifline[0]:
-                    data[(sascifline[0].split('.')[1])]=[]
-                if len(sascifline)>2 and len(all_lines[indx-1])>0 and'scan_intensity' in all_lines[indx-1][0]:
-                    for indx_sub,sascifline_sub in enumerate(all_lines[indx:]):
-                        if len(sascifline_sub)>2 and '#' not in sascifline_sub and 'sas' not in sascifline_sub[0] :  
-                            for num,key in enumerate(list(data.keys())):
+            # Int_dict={}
+            all_lines = self.get_all_sascif(code)
+            data = {}
+            for indx, sascifline in enumerate(all_lines):
+                if len(sascifline) < 2 and len(sascifline) > 0 and 'scan_intensity' in sascifline[0]:
```

Comment:
`0 < len(sascifline) < 2` is more concise - or even just `len(sascifline) == 1`.